src/profile-logic/import/simpleperf.js

```diff
@@ -0,0 +1,644 @@
+      // their index by other tables.
+      stringArray: this.strings.serializeToArray(),
+      funcTable: this.funcTable.toJson(),
+      resourceTable: this.resourceTable.toJson(),
+      nativeSymbols: {
+        libIndex: [],
+        address: [],
+        name: [],
+        functionSize: [],
+        length: 0,
+      },
+    };
+  }
+
+  enableCpuClock(cpuClockEventId: number): void {
+    this.cpuClockEventId = cpuClockEventId;
+
+    if (cpuClockEventId >= 0) {
+      this.sampleTable.weight = [];
+      this.sampleTable.weightType = 'tracing-ms';
+    }
+  }
+
+  addSample(sample: report.ISample, fileMap: Map<number, report.IFile>): void {
+    let prefixStackId: number | null = null;
+    for (const frame of sample.callchain.reverse()) {
+      const file: report.IFile = fileMap.get(frame.fileId);
+
+      const resourceIndex = this.resourceTable.findOrAddResource(file);
+      const methodName =
+        frame.symbolId >= 0
+          ? file.symbol[frame.symbolId]
+          : `${file.path.split(/[\\/]/).pop()}+0x${frame.vaddrInFile.toString(16)}`;
+
+      const funcIndex = this.funcTable.findOrAddFunc(methodName, resourceIndex);
+
+      const filePath = file.path ?? '';
+      // const fileInAppData = filePath.startsWith("/data/app/");
+      const fileInSystem =
+        filePath.startsWith('/apex/') ||
+        filePath.startsWith('/system/') ||
+        filePath.startsWith('/vendor/');
+
+      let category: IndexIntoCategoryList = Categories.Other;
+      if (filePath === '[kernel.kallsyms]' || filePath.endsWith('.ko')) {
+        category = Categories.Kernel;
+      } else if (filePath.endsWith('.so')) {
+        category = fileInSystem ? Categories.System : Categories.Native;
+      } else if (
+        filePath === '[JIT app cache]' ||
+        filePath.endsWith('.vdex') ||
+        filePath.endsWith('.apk') ||
+        filePath.endsWith('.jar') ||
+        filePath.endsWith('.oat') ||
+        filePath.endsWith('.odex')
+      ) {
+        const isJavaMethod =
+          methodName.startsWith('java.') ||
+          methodName.startsWith('javax.') ||
+          methodName.startsWith('kotlin.') ||
+          methodName.startsWith('kotlinx.') ||
+          methodName.startsWith('dalvik.');
+        const isAndroidMethod =
+          methodName.startsWith('android.') ||
+          methodName.startsWith('com.android.') ||
+          methodName.startsWith('androidx.') ||
+          methodName.startsWith('libcore.');
+        category =
+          fileInSystem || isAndroidMethod || isJavaMethod
+            ? Categories.System
+            : Categories.Java;
+      }
+
+      const frameIndex = this.frameTable.findOrAddFrame(funcIndex, category);
+
+      prefixStackId = this.stackTable.findOrAddStack(
+        frameIndex,
+        prefixStackId,
+        category
+      );
+    }
+
+    this.sampleTable.stack.push(prefixStackId);
+    $NotNull(this.sampleTable.time).push(toMilliseconds(sample.time ?? 0));
```

Comment:
$NotNull is probably not needed if you use a Thread instead of a SerializedThread